pkgs/development/libraries/libpointmatcher/default.nix

```diff
@@ -0,0 +1,41 @@
+{stdenv, fetchFromGitHub, cmake, eigen, boost, libnabo}:
+
+let inherit (stdenv.lib) getVersion versionAtLeast; in
+
+assert versionAtLeast (getVersion eigen) "3";
```

Comment:
Something like that, or in the top-level package set do
```
myPackage = callPackage { foo = foo_21; }
```

Checking version strings is appropriate some times, but ought to be used sparingly.